libs/server/Resp/KeyAdminCommands.cs

```diff
@@ -2,14 +2,228 @@
+            {
+                while (!RespWriteUtils.WriteDirect(CmdStrings.RESP_ERRNOTFOUND, ref dcurr, dend))
+                    SendAndReset();
+                return true;
+            }
+
+            Span<byte> encodedLength = stackalloc byte[5];
+
+            if (!RespLengthEncodingUtils.TryWriteLength(value.ReadOnlySpan.Length, encodedLength, out var bytesWritten))
+            {
+                while (!RespWriteUtils.WriteError("ERR DUMP payload length is invalid", ref dcurr, dend))
+                    SendAndReset();
+                return true;
+            }
+
+            encodedLength = encodedLength.Slice(0, bytesWritten);
+
+            // Len of the dump (payload type + redis encoded payload len + payload len + rdb version + crc64)
+            var len = 1 + encodedLength.Length + value.ReadOnlySpan.Length + 2 + 8;
+            Span<byte> lengthInASCIIBytes = stackalloc byte[NumUtils.NumDigitsInLong(len)];
+            var lengthInASCIIBytesLen = NumUtils.LongToSpanByte(len, lengthInASCIIBytes);
+
+            // Total len (% + length of ascii bytes + CR LF + payload type + redis encoded payload len + payload len + rdb version + crc64 + CR LF)
+            var totalLength = 1 + lengthInASCIIBytesLen + 2 + 1 + encodedLength.Length + value.ReadOnlySpan.Length + 2 + 8 + 2;
+
+            byte[] rentedBuffer = null;
+            var buffer = totalLength <= 256
+                ? stackalloc byte[256]
+                : (rentedBuffer = ArrayPool<byte>.Shared.Rent(totalLength));
+
+            var offset = 0;
+
+            // Write RESP bulk string prefix and length
+            buffer[offset++] = 0x24; // '$'
+            lengthInASCIIBytes.CopyTo(buffer[offset..]);
+            offset += lengthInASCIIBytes.Length;
+            buffer[offset++] = 0x0D; // CR
+            buffer[offset++] = 0x0A; // LF
+
+            // value type byte
+            buffer[offset++] = 0x00;
+
+            // length of the span
+            encodedLength.CopyTo(buffer[offset..]);
+            offset += encodedLength.Length;
+
+            // copy value to buffer
+            value.ReadOnlySpan.CopyTo(buffer[offset..]);
+            offset += value.ReadOnlySpan.Length;
+
+            // Write RDB version
+            buffer[offset++] = (byte)(RDB_VERSION & 0xff);
+            buffer[offset++] = (byte)((RDB_VERSION >> 8) & 0xff);
+
+            // Compute and write CRC64 checksum
+            var payloadToHash = buffer.Slice(1 + lengthInASCIIBytes.Length + 2 + 1,
+                encodedLength.Length + value.ReadOnlySpan.Length + 2);
+
+            var crcBytes = Crc64.Hash(payloadToHash);
+            crcBytes.CopyTo(buffer[offset..]);
+            offset += crcBytes.Length;
+
+            // Write final CRLF
+            buffer[offset++] = 0x0D; // CR
+            buffer[offset] = 0x0A; // LF
+
+            while (!RespWriteUtils.WriteDirect(buffer.Slice(0, totalLength), ref dcurr, dend))
+                SendAndReset();
```

Comment:
For large objects (when the output is bigger than output buffer) `WriteDirect/SendAndReset` will fail. You can use `WriteDirectLarge` method in RespServerSession.cs file but it's currently private, you have to make it public. @TalZaccai Please provide your thoughts on making it `public` or suggest any alternative method.

Add a test case to validate dump command for large objects